### DISCUSSION TITLE
Get logged UserId on lifecycle hooks

### DISCUSSION THREAD

**Discussion by kennethsato** [mikro-orm/mikro-orm#2911] [Q&A] [answered]
Hello

Is it possible to get logged user id inside a lifecycle hook?
I'm currently using afterUpdate hook. I want to log changes on one of my entities and I'd like to save which user did that change.
Is it possible to send it as an optional parameter on persistAndFlush or something like that?

```
  @AfterUpdate()
  async afterUpdate(
    eventArgs: EventArgs<Insercao>,
  ): Promise<void> {
    ... code here
}
```

**Comment by B4nan** [comment, marked as answer]
You can use filter params, even if you dont want to use filters.

https://mikro-orm.io/docs/filters#global-filters

**Reply by kennethsato** [reply]
Thank you again, @B4nan.

It worked.
An example if someone stuck at the same point:

I've setFilterParams in my Entity Manager on my service:

`this.databaseProvider.getConnection().em.setFilterParams('insercao', { user_id: decodedUserId });`


Then I just get it at my lifecycle hook (afterUpdate on my case):

`      const user_id = eventArgs.em.getFilterParams('insercao');`

ps: Insercao is my entity name.

**Reply by kennethsato** [reply]
Hello @B4nan 

Was there any significant change on this on v5-v6 Upgrade?

I'm trying to update, but when trying to setFilterParams, I'm getting the following error:

UnhandledPromiseRejection: This error originated either by throwing inside of async function without a catch block, or by rejecting a promise which was noandled with .catch(). The promise rejected with the reason "#<AppError>".
    at throwUnhandledRejectionsMode (node:internal/process/promises:389:7)
    at processPromiseRejections (node:internal/process/promises:470:17)
    at processTicksAndRejections (node:internal/process/task_queues:96:32)
[ERROR] 11:57:26 UnhandledPromiseRejection: This error originated either by twing inside of an async function without a catch block, or by rejecting a proe which was not handled with .catch(). The promise rejected with the reason "ppError>".

If I comment setFilterParams line, the error goes away, and obviously I get undefined for getFilterParams on my lifecycle hook:
const usuario_id = eventArgs.em.getFilterParams('chamada').usuario_id;

**Reply by B4nan** [reply]
Well, of course there were many changes, it was a major release, and there were many other changes in feature releases that followed v6.0. Hard to guess what is causing this without a reproduction, either way, if you get unhandled rejection, you must be missing an await somewhere, if you fix that, you should get a better understanding on where/why this happens. Try to get the actual stack trace first, this might help:

```ts
process.on('unhandledRejection', (reason, p) => {
  console.log('Unhandled Rejection at: Promise', p, 'reason:', reason);
});
```

**Reply by kennethsato** [reply]
Ops... Sorry!. And thanks again @B4nan !!
I thought the errror was related to setFilterParams, but it was not.
I was able to follow the stack trace and found a Repository.persistAndFlush that I hadn't replaced for em.persistAndFlush yet
